control-plane/agents/src/bin/core/main.rs

```diff
@@ -88,13 +88,38 @@ pub(crate) struct CliArgs {
     /// If `None` do not limit the number of rebuilds.
     #[clap(long)]
     max_rebuilds: Option<NumRebuilds>,
+
+    #[clap(flatten)]
+    thin_args: ThinArgs,
 }
 impl CliArgs {
     fn args() -> Self {
         CliArgs::parse()
     }
 }
 
+/// Cluster wide thin provisioning parameters.
+#[derive(Debug, clap::Parser, Clone)]
+pub(crate) struct ThinArgs {
+    /// The allowed pool commitment limit when dealing with thin provisioned volumes.
+    /// Example: If the commitment is 250 and the pool is 10GiB we can overcommit the pool
+    /// up to 25GiB (create 2 10GiB and 1 5GiB volume) but no further.
+    #[clap(long, env = "POOL_COMMITMENT", default_value = "250")]
+    pool_commitment: u64,
+    /// When creating pool replicas for an existing volume, each replica pool must have at least
+    /// this much free space percentage of the volume size.
+    /// Example: if this value is 40 and the pool has 40GiB free, then the max volume size allowed
+    /// to be created on the pool is 100GiB.
+    #[clap(long, env = "VOLUME_COMMITMENT_INITIAL", default_value = "40")]
+    volume_commitment_initial: u64,
+    /// When creating pool replicas for an existing volume, each replica pool must have at least
+    /// this much free space percentage of the volume size.
+    /// Example: if this value is 40, the pool has 40GiB free, then the max volume size allowed
+    /// to be created on the pool is 100GiB.
```

Comment:
Q: Do these parameters represent the same thing? The comments are identical.

Reply:
Sigh... they're identical because I messed up the rebase... I'll fix that up next week.